modules/internal/String.chpl

```diff
@@ -602,11 +602,27 @@ module String {
   }
 
   pragma "no doc"
-  proc chpl_createStringWithLiteral(x: c_string, length: int, numCodepoints: int) {
+  proc chpl_createLiteralsBuffer(size: int): c_void_ptr {
+    return c_malloc(uint(8), size):c_void_ptr;
```

Comment:
What's the challenge for freeing this? I would imagine putting this allocation in a module level variable and freeing that in module deinitializer should work, but do we end up deallocating the buffer too early if we do that?

Reply:
I would also expect it to work but I'd like to try that in a different PR at the very least. I'm open to feedback for whether or not we should try to do it for this release. Anyway I am worried that we might run in to an issue but it should be easy enough to try what you are describing.

Reply:
As a generally cautious person around the release dates, I feel brave about trying to do this for the release. A full ASAN test with local and gasnet configs would make me feel more comfortable about it. And if that testing turns up some bugs, we can punt it.

That being said, this leak is not too worrisome to me, as it is not something that would grow dynamically or anything. It'll be an O(n) leak where n is the amount of string literals used in the code. And we'll be freeing it at the teardown of the application, so it shouldn't impact the "steady state" memory footprint of the application in any way.

So, I say we go for it if you have the time. But I'm happy with whatever you decide at the end.

Reply:
I've created #18319 for this.